### PR TITLE
Round Button Component

### PR DESCRIPTION
**Pull request by mormo17**

### CODE REVIEW

File: Chamasma/Common/Sources/Common/Extensions/Color+Extension.swift [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            green = CGFloat((rgb & 0x00FF0000) >> 16) / 255.0
+            blue = CGFloat((rgb & 0x0000FF00) >> 8) / 255.0
+            opacity = CGFloat(rgb & 0x000000FF) / 255.0
+        } else {
+            return nil
+        }
+
+        self.init(red: red, green: green, blue: blue, opacity: opacity)
+    }
+}
```
Comment (Khizanag) [inline Chamasma/Common/Sources/Common/Extensions/Color+Extension.swift:47]: Please branch from master branch for future PRs

File: Chamasma/DesignSystem/Sources/DesignSystem/Components/ComponentViews/Buttons/RoundButton/RoundButton.swift [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+struct RoundButton: View {
+    let model: RoundButtonModel
+
+    init(model: RoundButtonModel) {
+        self.model = model
+    }
+
+    var body: some View {
+        Button {
+            print("ra")
```
Comment (Khizanag) [inline Chamasma/DesignSystem/Sources/DesignSystem/Components/ComponentViews/Buttons/RoundButton/RoundButton.swift:20]: What?

File: Chamasma/DesignSystem/Sources/DesignSystem/Components/ComponentViews/Buttons/RoundButton/RoundButtonModel.swift [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+import Common
+import SwiftUI
+
+struct RoundButtonModel {
+    let image: RoundButtonImage
+    let state: RoundButtonState
+    let size: RoundButtonSize
+
+    init(
```
Comment (Khizanag) [inline Chamasma/DesignSystem/Sources/DesignSystem/Components/ComponentViews/Buttons/RoundButton/RoundButtonModel.swift:16]: No init is needed

File: Chamasma/DesignSystem/Sources/DesignSystem/Components/ComponentViews/Buttons/RoundButton/RoundButtonModel.swift [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+}
+
+extension RoundButtonModel {
+    enum RoundButtonImage {
+        case plus
+        case search
+        case backArrow
+        case close
+
+        var name: Image {
```
Comment (Khizanag) [inline Chamasma/DesignSystem/Sources/DesignSystem/Components/ComponentViews/Buttons/RoundButton/RoundButtonModel.swift:34]: Name should not return Image

File: Chamasma/DesignSystem/Sources/DesignSystem/Components/ComponentViews/Buttons/RoundButton/RoundButtonModel.swift [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+                return Color(hex: "4ED199") ?? .black
+            }
+        }
+    }
+
+    enum RoundButtonSize {
+        case large
+        case normal
+
+        var value: CGFloat {
```
Comment (Khizanag) [inline Chamasma/DesignSystem/Sources/DesignSystem/Components/ComponentViews/Buttons/RoundButton/RoundButtonModel.swift:75]: Can we use Double instead?

File: Chamasma/DesignSystem/Sources/DesignSystem/Components/ComponentViews/Buttons/RoundButton/RoundButtonModel.swift [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+            switch self {
+            case .large:
+                return 100
+            case .normal:
+                return 40
+            }
+        }
+
+        var font: CGFloat {
```
Comment (Khizanag) [inline Chamasma/DesignSystem/Sources/DesignSystem/Components/ComponentViews/Buttons/RoundButton/RoundButtonModel.swift:84]: Font named property should return font

File: Chamasma/DesignSystem/Sources/DesignSystem/Components/ComponentViews/Buttons/RoundButton/RoundButtonModel.swift [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+            switch self {
+            case .large:
+                return 50
+            case .normal:
+                return 20
```
Comment (Khizanag) [inline Chamasma/DesignSystem/Sources/DesignSystem/Components/ComponentViews/Buttons/RoundButton/RoundButtonModel.swift:89]: Giving appropriate size to view is Views responsibility I think

**Review by Khizanag — COMMENTED** [review]
Sizing

File: Chamasma/Common/Sources/Common/Extensions/Color+Extension.swift (hunk 1 above)
Comment (mormo17) [reply Chamasma/Common/Sources/Common/Extensions/Color+Extension.swift:47]: yeah, you are right, but unfortunately I needed those changes in this component

File: Chamasma/DesignSystem/Sources/DesignSystem/Components/ComponentViews/Buttons/RoundButton/RoundButton.swift (hunk 2 above)
Comment (mormo17) [reply Chamasma/DesignSystem/Sources/DesignSystem/Components/ComponentViews/Buttons/RoundButton/RoundButton.swift:20]: you mean ra?

File: Chamasma/DesignSystem/Sources/DesignSystem/Components/ComponentViews/Buttons/RoundButton/RoundButtonModel.swift (hunk 4 above)
Comment (mormo17) [reply Chamasma/DesignSystem/Sources/DesignSystem/Components/ComponentViews/Buttons/RoundButton/RoundButtonModel.swift:34]: can you, please, suggest better name? I have been thinking and was not able to figure it out

File: Chamasma/DesignSystem/Sources/DesignSystem/Components/ComponentViews/Buttons/RoundButton/RoundButtonModel.swift (hunk 5 above)
Comment (mormo17) [reply Chamasma/DesignSystem/Sources/DesignSystem/Components/ComponentViews/Buttons/RoundButton/RoundButtonModel.swift:75]: why? this is used in frame(width: _, height: _) which uses CGFloat as I saw

File: Chamasma/DesignSystem/Sources/DesignSystem/Components/ComponentViews/Buttons/RoundButton/RoundButtonModel.swift (hunk 6 above)
Comment (mormo17) [reply Chamasma/DesignSystem/Sources/DesignSystem/Components/ComponentViews/Buttons/RoundButton/RoundButtonModel.swift:84]: can you please suggest better one?

File: Chamasma/DesignSystem/Sources/DesignSystem/Components/ComponentViews/Buttons/RoundButton/RoundButtonModel.swift (hunk 7 above)
Comment (mormo17) [reply Chamasma/DesignSystem/Sources/DesignSystem/Components/ComponentViews/Buttons/RoundButton/RoundButtonModel.swift:89]: I will think how to refactor it, thanks

File: Chamasma/Common/Sources/Common/Extensions/Color+Extension.swift (hunk 1 above)
Comment (mormo17) [reply Chamasma/Common/Sources/Common/Extensions/Color+Extension.swift:47]: please, if you think that I can resolve conversations, please resolve it, or comment that I can, or comment what should I do in order to resolve

File: Chamasma/DesignSystem/Sources/DesignSystem/Components/ComponentViews/Buttons/RoundButton/RoundButtonModel.swift (hunk 5 above)
Comment (mormo17) [reply Chamasma/DesignSystem/Sources/DesignSystem/Components/ComponentViews/Buttons/RoundButton/RoundButtonModel.swift]: can I resolve it or should I change sth?

**Pull request merged**